sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -297,6 +298,27 @@ public void setDecimal(int ordinal, Decimal value, int precision) {
     }
   }
 
+  @Override
+  public void setInterval(int ordinal, CalendarInterval value) {
+    assertIndexIsValid(ordinal);
+    long cursor = getLong(ordinal) >>> 32;
+    assert cursor > 0 : "invalid cursor " + cursor;
+    if (value == null) {
+      setNullAt(ordinal);
+      // zero-out the bytes
+      Platform.putInt(baseObject, baseOffset + cursor, 0);
+      Platform.putInt(baseObject, baseOffset + cursor + 4, 0);
```

Comment:
we don't need to put int twice: `Platform.putLong(baseObject, baseOffset + cursor, 0L);`